sharding/nodesCoordinator/indexHashedNodesCoordinatorLite_test.go

```diff
@@ -0,0 +1,89 @@
+package nodesCoordinator
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIndexHashedNodesCoordinator_SetNodesConfigFromValidatorsInfo(t *testing.T) {
+	t.Parallel()
+
+	arguments := createArguments()
+
+	ihgs, err := NewIndexHashedNodesCoordinator(arguments)
+	require.Nil(t, err)
+	epoch := uint32(1)
+
+	randomness := []byte("rand seed")
+
+	ihgs.nodesConfig[epoch] = ihgs.nodesConfig[0]
+
+	body := createBlockBodyFromNodesCoordinator(ihgs, epoch)
+	validatorsInfo, err := createValidatorInfoFromBody(body, arguments.Marshalizer, 10)
+	require.Nil(t, err)
+
+	err = ihgs.SetNodesConfigFromValidatorsInfo(epoch, randomness, validatorsInfo)
+
+	validators, err := ihgs.GetAllEligibleValidatorsPublicKeys(epoch)
+	require.Nil(t, err)
+	require.NotNil(t, validators)
+}
+
+func TestIndexHashedNodesCoordinator_SetNodesConfigFromValidatorsInfoMultipleEpochs(t *testing.T) {
+	t.Parallel()
+
+	arguments := createArguments()
+
+	ihgs, err := NewIndexHashedNodesCoordinator(arguments)
+	require.Nil(t, err)
+	epoch := uint32(1)
+
+	randomness := []byte("rand seed")
+
+	ihgs.nodesConfig[epoch] = ihgs.nodesConfig[0]
+
+	body := createBlockBodyFromNodesCoordinator(ihgs, epoch)
+	validatorsInfo, err := createValidatorInfoFromBody(body, arguments.Marshalizer, 10)
+	require.Nil(t, err)
+
+	ihgs.SetNodesConfigFromValidatorsInfo(epoch, randomness, validatorsInfo)
+	ihgs.SetNodesConfigFromValidatorsInfo(epoch+1, randomness, validatorsInfo)
+
+	_, ok := ihgs.nodesConfig[epoch+1]
+	assert.True(t, ok)
+
+	ihgs.SetNodesConfigFromValidatorsInfo(epoch+4, randomness, validatorsInfo)
```

Comment:
If you change `nodesCoordinatorStoredEpochs` to let's say, 10, this test would fail. 
Perhaps instead of `epoch+4` you could `epoch+nodesCoordinatorStoredEpochs`

You could also check for `epoch+1` uif it is still stored

Reply:
:+1: